### PR TITLE
Some improvements to run windows in a factory

### PR DESCRIPTION
**Pull request by mgax**
* Connect an ISO file as cdrom (or two, because windows iso + virtio drivers iso)
* Simplify arm64 code (purely as side-effect, but it's so much cleaner now!)
* Connect to the serial console on stdio (which deactivates the whole connect-to-guest-via-ssh dance. windows has no ssh.)
* Create VNC server (because next next finish.)
* Save vm's disk back into the original image with `--commit`
* Commands to `export` and `import` images (need docs)
* Replace the weird `--platform` argument with an `--image` argument that fits in with `--memory` or `--share`

### CODE REVIEW

**Review by gabriel-v — COMMENTED** [review]
arm64 build: 
`+ ./factory run true` gives out 
`RuntimeError: VM did not create its sockets`

see https://jenkins.liquiddemo.org/blue/organizations/jenkins/liquidinvestigations%2Ffactory/detail/PR-24/2/pipeline/30

**Comment by mgax** [comment]
Yeah, that's weird, running that by hand works. Still digging.

**Comment by mgax** [comment]
@gabriel-v I've rebased onto master and now it builds. I think the previous failure was a fluke.

File: Readme.md [hunk 1]
```diff
@@ -80,3 +78,86 @@ Both modes support the following options:
...
     ./factory login --udp 1053:53
     ```
+
+* `--vnc`: open a VNC server. The port number must be between 5900 and 5999.
+
+    ```
+    ./factory login --vnc 5900
+    ```
+
+* `--commit`: after the VM quits, save the changes to its disk.
```
Comment (gabriel-v) [inline Readme.md]: You're saying this is interactive (asks for confirmation when the VM dies before overwriting). Where's the `-y` or `--always-commit` or something? Also, document it here.

File: factory.py [hunk 2]
```diff
@@ -283,24 +318,74 @@ def vm_bootstrap(self, timeout=60):
...
             finally:
                 kill_qemu_via_qmp('vm.qmp')
 
     @staticmethod
     def invoke_ssh(cmd):
         subprocess.run(cmd, check=True)
 
+    @staticmethod
+    def invoke_console(socket_path):
+        subprocess.run(['socat', '-,cfmakeraw,escape=0xf', str(socket_path)])
```
Comment (gabriel-v) [inline factory.py:392]: I don't see `socat` listed as a dependency in the readme. Please add it.

File: factory.py [hunk 3]
```diff
@@ -317,13 +402,20 @@ def ssh(self, cmd=None):
...
+    def console(self):
+        self.invoke_console(self.var / 'vm.mon')
+
 
 def add_vm_arguments(parser):
+    parser.add_argument('-i', '--image', default=DEFAULTS[get_arch()])
     parser.add_argument('--share', action='append', default=[])
     parser.add_argument('-m', '--memory', default=512, type=int)
-    parser.add_argument('-p', '--smp', default=1, type=int)
+    parser.add_argument('-s', '--smp', default=1, type=int)
```
Comment (gabriel-v) [inline factory.py:418]: Sure, change them all you want, they're not used or documented anywhere anyway. I'd even say we'd be better off removing all the short options here.

**Review by gabriel-v — CHANGES_REQUESTED** [review]
Please add some auto-confirmation to `--commit` to make it scriptable.

**Pull request merged**